File: bot.lua
JSON = (loadfile "JSON.lua")()
socket = require"socket"
socket.unix = require"socket.unix"

now = os.time()

function on_msg_receive (msg)
	if msg.date < now then
		return
	end
	if msg.out then
		return
	end

	print(msg)

	if msg.text then
		c = socket.unix()
		c:connect("socket")
		c:send(JSON:encode(msg) .. "\r\n\r\n")

		while 1 do
			data = ""
			while 1 do
				if string.find(data, "\r\n\r\n") then
					break
				end

				tmp = c:receive(1)
				if (tmp == "") or tmp == nil then
					break
				end

				data = data .. tmp
			end
			data = JSON:decode(data)

			print ('action : ' .. data['action'])
			if (data['action'] == 'send_msg') then
				send_msg(get_receiver(msg), data['arg'], ok_cb, false)
			end

			if (data['action'] == 'send_img') then
				send_photo(get_receiver(msg), data['arg'], ok_cb, false)
			end

			if (data['action'] == 'bye') then
				break
			end
		end

		c:close()
	end

	mark_read(get_receiver(msg))
end
 
function on_our_id (id)
end
 
function on_secret_chat_created (peer)
end
 
function on_user_update (user)
end
 
function on_chat_update (user)
end
 
function on_get_difference_end ()
end
 
function on_binlog_replay_end ()
end

function get_receiver(msg)
  if msg.to.type == 'user' then
    return 'user#id'..msg.from.id
  end
  if msg.to.type == 'chat' then
    return 'chat#id'..msg.to.id
  end
end

function ok_cb(extra, success, result)
end

File: bot.py
#!/usr/bin/python
#-*-coding:utf-8 -*-

import socket
import sys
import os
import time
import random
import json
import requests

#init variables

reload(sys)
sys.setdefaultencoding("utf-8")

def write(text):
    c.send(json.dumps({'action':'send_msg', 'arg': text}) + "\r\n\r\n")

def img_write(text):
    c.send(json.dumps({'action':'send_img', 'arg': text}) + "\r\n\r\n")

server_address = './socket'
try:
    os.unlink(server_address)
except OSError:
    if os.path.exists(server_address):
        raise

sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
sock.bind(server_address)
sock.listen(1)

while True:
    c, ca = sock.accept()
    c.settimeout(3)

    try:
        print 'got connection'
        t = time.time()

        data = ""
        while True:
            tmp = c.recv(1)
            data += tmp
            if "\r\n\r\n" in data:
                break

        print 'received'
        data = json.loads(data)
        print time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(int(data['date'])+32400)) + " " + data['from']['print_name'] + " : " + data['text'] 
        msgArr = data['text'].split()
        #명령어들
        if msgArr[0] == "TEST":
            write("TEST")
		
        c.send(json.dumps({'action':'bye'}) + "\r\n\r\n")

    except socket.error:
        print 'socket error'
        continue
    finally:
        c.close()
